package-res/resources/web/prompting/PromptPanel.js

```diff
@@ -1391,6 +1493,87 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
          */
         setBlockUiOptions: function(options) {
           this.getDashboard()._setBlockUiOptions(options);
+        },
+
+        /**
+         * Gets a current state of the prompting system.
+         *
+         * @name PromptPanel#getState
+         * @method
+         * @returns {Object} The current state which consists of the next properties:
+         *                   <ul>
+         *                     <li>'promptNeeded' &lt;Boolean&gt; - True if prompts are needed, False otherwise (read only property)</li>
+         *                     <li>'paginate' &lt;Boolean&gt; - True if pagination is active, False otherwise (read only property)</li>
+         *                     <li>'totalPages' &lt;Number&gt; - The number of total pages of the report (read only property)</li>
+         *                     <li>'showParameterUI' &lt;Boolean&gt; - The boolean value of the parameter ShowParameters (read only property)</li>
+         *                     <li>'allowAutoSubmit' &lt;Boolean&gt; - The value of autoSubmit, or if it is undefined the value of autoSubmitUI (read only property)</li>
+         *                     <li>'parametersChanged' &lt;Boolean&gt; - True if the parameters have changed, False otherwise</li>
+         *                     <li>'autoSubmit' &lt;Boolean&gt; - True is the prompt is in auto submit mode, False otherwise</li>
+         *                     <li>'page' &lt;Number&gt; - The number of the page</li>
+         *                   </ul>
+         * @example
+         * var currentState = api.operation.state();
+         * // Return value:
+         * //   {
+         * //     "promptNeeded":false,
+         * //     "paginate":true,
+         * //     "totalPages":1,
+         * //     "showParameterUI":true,
+         * //     "allowAutoSubmit":false,
+         * //     "parametersChanged":false,
+         * //     "autoSubmit":false,
+         * //     "page":-1
+         * //   }
+         */
+        getState: function() {
+          var result = {
+            promptNeeded: this.paramDefn.promptNeeded,
+            paginate: this.paramDefn.paginate,
+            totalPages: this.paramDefn.totalPages,
+            showParameterUI: this.paramDefn.showParameterUI(),
+            allowAutoSubmit: this.paramDefn.allowAutoSubmit(),
+            parametersChanged: this.parametersChanged,
+            autoSubmit: this.paramDefn.autoSubmit,
+            page: this.paramDefn.page
+          };
+          return result;
+        },
+
+        /**
+         * Modifys a state of the prompting system.
+         *
+         * @name PromptPanel@setState
+         * @method
+         * @param {Object} [state]                    The set of flags which will be applyed to current state. It's optional parameter.
```

Comment:
@annapopova1  is the stage really optional? does it makes sense to call setState without the argument